### ISSUE TITLE
`DoubleNegation` default style (`allowed_in_returns`) should allow methods with rescue blocks

### ISSUE THREAD

**Issue by Drowze** [rubocop/rubocop#10010]
obs.: This is probably more a feature request than a bug - please let me know your thoughts.

## Expected behavior

Since RuboCop's `Style/DoubleNegation` cop default configuration (`allowed_in_returns`) allows this:
```ruby
def db_ok?
  !!db.query('1 + 1')
end
```

I would expect it to also allow this use of double negation:
```ruby
def safe_db_ok?
  !!db.query('1 + 1')
rescue DatabaseError
  false
end
```

That probably means to also allow this other usage:
```ruby
def safe_db_ok?
  !!db.query('1 + 1')
rescue DatabaseError
  !!ENV['DATABASE_URL']
end
```

## Actual behavior

RuboCop's `Style/DoubleNegation` flags the double negation usage on any method that contains a rescue block.

## RuboCop version

```
1.19.0 (using Parser 3.0.2.0, rubocop-ast 1.10.0, running on ruby 2.7.3 x86_64-darwin20)
```

**Comment by koic** [comment]
I think this is a false positive because it should be accepted as a similar semantics.
